src/citationnet/static/force-citationNet.js

```diff
@@ -0,0 +1,419 @@
+    */
+    constructor(jsondata = null) {
+        // if jsonPath is not provided try to get globJsonPath, show alert if fails
+        try {
+            if (!(jsondata)) jsondata = jsondata;
+        } catch (error) {
+            alert("no JSON containing citation data specified, graph cannot be displayed");
+        }
+
+        this.jsondata = jsondata;
+        this.is_initialized = false;
+    }
+
+    /**
+    * Fetches and processes data, initializes graph and sets view. Constructors cannot be async in JS, which is needed for fetching and saving data.
+    */
+    async initialize(make_cylinder = false) {
+        this.container = document.getElementById('3d-graph');
+        // fetch data (async)
+        // this.data = await fetchJSON(this.jsonPath);
+        // this.processData();
+        await this.getStats();
+        this.makeGraph(this.data);
+
+        // variables for control toggle-functions
+        this.nodeSize = false;
+        this.edgesOnlyInput = true;
+        this.distanceFromInputNode = false;
+
+        this.adaptWindowSize();
+        this.view('side');
+        this.toggleNodeSize();
+
+        if (make_cylinder) {
+            this.makeCylinder();
+            this.graph.controls()._listeners.change.push(this.renderLabels());
+        }
+
+        this.is_initialized = true;
+    }
+
+    /**
+    * Reads current window size, adapts canvas size and camera projection settings  to it.
+    */
+    adaptWindowSize() {
+        this.graph.height(window.innerHeight - document.getElementsByClassName("navbar")[0].scrollHeight);
+        this.graph.width(window.innerWidth);
+
+        this.graph.camera().left = this.graph.width() / -2;
+        this.graph.camera().right = this.graph.width() / 2;
+        this.graph.camera().top = this.graph.height() / 2;
+        this.graph.camera().bottom = this.graph.height() / -2;
+        this.graph.camera().updateProjectionMatrix();
+    }
+
+    /**
+    * Instantiate and configure graph object
+    * @returns {function} Brief description of the returning value here.
+    */
+    makeGraph() {
+        // sort nodes descending by data.nodes.attributes['ref-by-count'], get first items 'ref-by-count'-attribute
+        const maxCites = this.data.nodes.sort((a, b) => (a.attributes['ref-by-count'] > b.attributes['ref-by-count']) ? -1
+            : (a.attributes['ref-by-count'] < b.attributes['ref-by-count']) ? 1
+                : 0)[0].attributes['ref-by-count'];
+
+        // make Graph object
+        this.graph = ForceGraph3D({
+            "controlType": 'trackball',
+            // turn off antaliasing and alpha for improved performance
+            "rendererConfig": { antialias: false, alpha: false }
+        })(this.container)
+            .graphData({ nodes: this.data.nodes, links: this.data.edges })
+            .nodeId('id')
+            .nodeLabel(node => {
+                var doi = (typeof node.attributes.doi !== 'undefined') ? node.attributes.doi : node.id;
```

Comment:
I suggest using `const` (see comment on visDanymic.html). I will skip similar comments for other `var`s